### PR TITLE
Emails: Update copy and hide placeholders on Email Comparison page

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull requests updates the `Email Comparison` page to:

* Update the description of Google Workspace to match changes from [this pull request](https://github.com/Automattic/wp-calypso/pull/58186)
* Refer to mailboxes instead of users in the price (i.e. `/user /month billed annually`)
* Remove duplicate billing information for Google Workspace (i.e. `billed annually`)
* Remove placeholders that just duplicate the labels of each form field

Note this last change is also visible on the `Add New Mailboxes` and `Set Up Mailbox` pages since they use the same forms for Titan and Google Workspace.

##### Email Comparison

Before | After
------ | -----
image <br/><br/><br/><br/><br/><br/><br/><br/>| image

##### Add New Mailboxes (Titan)

Before | After
------ | -----
image | image

##### Add New Mailboxes (Google Workspace)

Before | After
------ | -----
image | image

##### Set Up Mailbox

Before | After
------ | -----
image | image

#### Testing instructions

1. Run `git checkout update/email-comparison-page` and start your server, or access a [live branch](https://github.com/Automattic/wp-calypso/pull/58193#issuecomment-971861463)
2. Log into a WordPress.com account that has Titan, Google Workspace, and a domain with no email
3. Open the [`Emails` page](http://calypso.localhost:3000/email)
4. Click the `Add Email` button to access the `Email Comparison` page
5. Assert that you see the changes mentioned above
6. Go back to the `Emails` page
7. Click a Titan account
8. Click the `Set up mailbox` button to access the `Set Up Mailbox` page
9. Assert that placeholders are no longer displayed
10. Go back, and click the `Add new mailboxes` option to access the `Add New Mailboxes` page
11. Assert that placeholders are no longer displayed
12. Go back to the `Emails` page
13. Click a Google Workspace account
14. Click the `Add new mailboxes` option to access the `Add New Mailboxes` page
15. Assert that placeholders are no longer displayed

### CODE REVIEW

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 1]
```diff
@@ -357,7 +357,7 @@ class EmailProvidersComparison extends Component {
 		const productIsDiscounted = hasDiscount( gSuiteProduct );
 		const monthlyPrice = getMonthlyPrice( gSuiteProduct?.cost ?? null, currencyCode );
 		const formattedPrice = productIsDiscounted
-			? translate( '{{fullPrice/}} {{discountedPrice/}} /user /month billed annually', {
+			? translate( '{{fullPrice/}} {{discountedPrice/}} /mailbox /month billed annually', {
```
Comment (a8ci18n) [inline client/my-sites/email/email-providers-comparison/index.jsx:382]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 2]
```diff
@@ -369,7 +369,7 @@ class EmailProvidersComparison extends Component {
 					comment:
 						'{{fullPrice/}} is the formatted full price, e.g. $20; {{discountedPrice/}} is the discounted, formatted price, e.g. $10',
 			  } )
-			: translate( '{{price/}} /user /month billed annually', {
+			: translate( '{{price/}} /mailbox /month billed annually', {
```
Comment (a8ci18n) [inline client/my-sites/email/email-providers-comparison/index.jsx:394]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

File: client/my-sites/email/email-providers-comparison/index.jsx [hunk 3]
```diff
@@ -499,7 +489,7 @@ class EmailProvidersComparison extends Component {
 			translate,
 		} = this.props;
 
-		const formattedPrice = translate( '{{price/}} /user /month billed monthly', {
+		const formattedPrice = translate( '{{price/}} /mailbox /month billed monthly', {
```
Comment (a8ci18n) [inline client/my-sites/email/email-providers-comparison/index.jsx:514]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6979612

Thank you @stephanethomas for including a screenshot in the description! This is really helpful for our translators.

**Comment by daledupreez** [comment]
> Expand the Google Workspace form by default so users can compare our two main solutions easily

This is a feature of the current design/UX that was a core part of the design. I think we should defer removing that as part of this PR, as I think it needs some dedicated discussion before we make a call on this front.

**Review by olaseni — DISMISSED** [review]
~~LGTM! All screens are verified~~
Is there a reason why we are expanding two forms?

**Comment by stephanethomas** [comment]
> Expand the Google Workspace form by default so users can compare our two main solutions easily

I've rebased the branch, and removed the commit responsible for that change 👍.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6979612).

**Pull request merged**